alarm_backend/app.py

```diff
@@ -13,7 +16,32 @@
 
 
 @app.route("/alarms")
-def root():
+def get_all_alarms():
     """Get all alarms"""
+    all_alarm = []
     alarms = Alarm.query.all()
-    return alarms
\ No newline at end of file
+    for a in alarms:
+        all_alarm.append(serialize(a))
+    return jsonify(all_alarm)
+
+
+@app.route("/alarms/add", methods=['POST'])
+def add_alarm():
+    response = request.json
+    hour = response['hour']
+    minute = response['minutes']
+    amPm = response['amPmOption']
+    type = response['type']
+    difficulty = response['difficulty']
+
+    new_alarm = Alarm(hour=hour, minute=minute, AMorPM=amPm, type=type, difficulty=difficulty)
+    db.session.add(new_alarm)
+    db.session.commit()
+    return "Success"
+
+@app.route("/alarms/<int:alarm_id>/delete", methods=['POST'])
```

Comment:
Here's an example to how this route (and other routes) can be improved. See if you can pin point the differences!
```
@app.route("/alarms/<int:alarm_id>", methods=['DELETE'])
def delete_alarm(alarm_id):
    try:
        alarm = Alarm.query.get_or_404(alarm_id)
        db.session.delete(alarm)
        db.session.commit()
        return make_response(jsonify({"message": "Alarm successfully deleted", "id": alarm_id}), 200)
    except Exception as e:
        return make_response(jsonify({"error": str(e)}), 500)
```